packages/flutter_tools/lib/src/base/build.dart

```diff
@@ -294,11 +304,17 @@ class AOTSnapshotter {
       '-o', appLib,
       assemblyO,
     ];
-    final RunResult linkResult = await _xcode.clang(linkArgs);
-    if (linkResult.exitCode != 0) {
+    RunResult result = await _xcode.clang(linkArgs);
+    if (result.exitCode != 0) {
       _logger.printError('Failed to link AOT snapshot. Linker terminated with exit code ${compileResult.exitCode}');
+    } else if (splitDebugInfo != null) {
+      final String dSYMPath = _fileSystem.path.join(splitDebugInfo, targetArch, 'App.dSYM');
```

Comment:
I believe `App.framework.dSYM` is more idiomatic for framework dSYMs.
```suggestion
      final String dSYMPath = _fileSystem.path.join(splitDebugInfo, targetArch, 'App.framework.dSYM');
```

Reply:
made the change